packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -98,6 +98,35 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
     scrollRef
   });
 
+  const cachedFocusedKey = useRef(null);
+  const cachedCollection = useRef(null);
+  useEffect(() => {
+    if (
+      state.collection.size > 0 &&
+      state.selectionManager.focusedKey === null &&
+      cachedFocusedKey.current !== null
+    ) {
+      const node = cachedCollection.current.getItem(cachedFocusedKey.current);
+      const isCell = node.type === 'cell' || node.type === 'rowheader' || node.type === 'column';
+      const parentNode =
+        node.parentKey && isCell ?
+        cachedCollection.current.getItem(node.parentKey) :
+        cachedCollection.current.getItem(node.key);
+      const rows = state.collection.rows;
+      const newRow = parentNode.index < rows.length ? rows[parentNode.index] : rows[rows.length - 1];
+      const keyToFocus =
+        newRow && newRow.hasChildNodes && isCell ?
+        [...newRow.childNodes][node.index].key :
+        newRow.key;
```

Comment:
We probably do need to skip over disabled rows in case the new row at that index can't be focused. It'll need to depend on the `disabledBehavior` since some items only disable selection and still allow focus - SelectionManager's `isDisabled` method should handle that.